accounts/models.py

```diff
@@ -641,7 +641,19 @@ def get_stats_for_profile_page(self):
             cache.set(settings.USER_STATS_CACHE_KEY.format(self.user_id), stats_from_cache, 60*60*24)
         stats_from_db.update(stats_from_cache)
         return stats_from_db
-
+    
+    def get_ai_preference(self):
+        try:
+            return self.user.ai_preference.preference
+        except AIPreference.DoesNotExist:
+            # If no preference is set, return the default one
+            return AIPreference.DEFAULT_AI_PREFERENCE
+        
+    def set_ai_preference(self, preference_value):
```

Comment:
isn't this just `create_or_update`?
I don't think the `.update` is correct, you need to filter on the user first